go-controller/hybrid-overlay/pkg/windows/service/service.go

```diff
@@ -0,0 +1,107 @@
+// +build windows
+
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+Based on https://github.com/kubernetes/kubernetes/blob/master/pkg/windows/service/service.go
+The reason for making a copy is that packages in kubernetes/kubernetes
+are not setup to be imported as modules without vendoring and pinning the sub
+components using the "require" directive. If more packages need to be imported
+then we can revisit importing this module too.
+*/
+
+package service
+
+import (
+	"os"
+	"time"
+
+	"k8s.io/klog"
+
+	"golang.org/x/sys/windows"
+	"golang.org/x/sys/windows/svc"
+)
+
+var (
+	service *handler
+)
+
+type handler struct {
+	fromsvc chan error
+}
+
+// InitService is the entry point for running the daemon as a Windows
+// service. It returns an indication of whether it is running as a service;
+// and an error.
+func InitService(serviceName string) error {
+	h := &handler{
+		fromsvc: make(chan error),
+	}
+
+	service = h
```

Comment:
@mansikulkarni96 where is `service` used? I can't find it.

Reply:
@dcbw not used anywhere here, this file was mostly referred from  [https://github.com/kubernetes/kubernetes/blob/master/pkg/windows/service/service.go](https://github.com/kubernetes/kubernetes/blob/master/pkg/windows/service/service.go) as we found it would be better to copy it instead of importing which would bring in unnecessary kubernetes imports. Some options in the file are added for future extensibility. But I guess it can be removed if we dont need in this case.

Reply:
@mansikulkarni96 yeah, I'll have comments on that later, we should clean it up to a minimal set.

Reply:
updated upstream and downstream